src/execute_linter.cc

```diff
@@ -13,55 +13,157 @@
+    for (std::string check_name : names) {
+      // The name inside of parantheses is stored in 'check_name'
+      // If it is not valid add error, otherwise enable/disable position
+      if (!error_map.count(check_name)) {
+        result.Add(ErrorCode::kNoLint, sql, position,
+                   absl::StrCat("Unkown NOLINT error category: ", check_name));
+      } else {
+        const ErrorCode& code = error_map[check_name];
+        if (type == "NOLINT")
+          options->Disable(code, position);
+        else
+          options->Enable(code, position);
+      }
+    }
+  }
   return result;
 }
 
-}  // namespace
-}  // namespace zetasql::linter
+// This function will parse "NOLINT"(<CheckName>) syntax from a sql file.
+// If NOLINT usage errors occur, it counts as a lint
+// error and it will be returned in a result.
+LinterResult ParseNoLintComments(absl::string_view sql,
+                                 LinterOptions* options) {
+  LinterResult result;
+  for (int i = 0; i < static_cast<int>(sql.size()); ++i) {
+    if (sql[i] == '\'' || sql[i] == '"') {
+      // The sql is inside string. This will ignore sql part until
+      // the same type(' or ") of character will occur
+      // without \ in the beginning. For example 'a"b' is a valid string.
+      for (int j = i + 1; j < static_cast<int>(sql.size()); ++j) {
+        if (sql[j - 1] == '\\' && sql[j] == sql[i]) continue;
+        if (sql[j] == sql[i] || j + 1 == static_cast<int>(sql.size())) {
+          i = j;
+          break;
+        }
+      }
+      continue;
+    }
 
-int main(int argc, char* argv[]) {
-  if (argc < 1) {
-    std::cout << "Usage: execute_linter < <file_name>\n" << std::endl;
-    return 1;
-  }
+    // Ignore multiline comments.
+    if (i > 0 && sql[i - 1] == '/' && sql[i] == '*') {
+      // It will start checking after '/*' and after the iteration
+      // finished, the pointer 'i' will be just after '*/' (incrementation
+      // from the for statement is included).
+      i += 2;
+      while (i < static_cast<int>(sql.size()) &&
+             !(sql[i - 1] == '*' && sql[i] == '/')) {
+        ++i;
+      }
+    }
 
-  std::string str;
-  for (std::string line; std::getline(std::cin, line);) {
-    str += line + "\n";
+    bool is_line_comment = false;
+    if (i > 0 && sql[i - 1] == '-' && sql[i] == '-') is_line_comment = true;
+    if (i > 0 && sql[i - 1] == '/' && sql[i] == '/') is_line_comment = true;
+    if (sql[i] == '#') is_line_comment = true;
+
+    if (is_line_comment) {
+      std::string line = "";
+      while (i + 1 < static_cast<int>(sql.size()) &&
+             sql[i] != options->LineDelimeter())
+        line += sql[++i];
+      result.Add(ParseNoLintSingleComment(line, sql, options, i));
+      continue;
+    }
   }
-  zetasql::linter::LinterResult result =
-      zetasql::linter::RunChecks(absl::string_view(str));
 
-  result.PrintResult();
+  return result;
+}
+
+LinterOptions GetOptionsFromConfig() {
```

Comment:
How about ReadOptionsFromFile?

Reply:
Yes, we will read options from config.proto file, but i didn't include it in this PR (this already large enough :p) It will be on seperate PR named 'proto'